src/lib/WalletInfoCollector.ts

```diff
@@ -197,16 +232,17 @@ export default class WalletInfoCollector {
         };
     }
 
-    private static _keyguardClient?: KeyguardClient; // TODO avoid the need to create another KeyguardClient here
-    private static _networkInitializationPromise?: Promise<void>;
-
-    private static _initializeDependencies(walletType: WalletType, checkLegacyActivity: boolean): void {
+    private static _initializeDependencies(walletType: WalletType): void {
         WalletInfoCollector._networkInitializationPromise =
             WalletInfoCollector._networkInitializationPromise
             || NetworkClient.createInstance(Config.networkEndpoint).init();
         WalletInfoCollector._networkInitializationPromise
             .catch(() => delete WalletInfoCollector._networkInitializationPromise);
-        if (walletType !== WalletType.BIP39 && (walletType !== WalletType.LEGACY || !checkLegacyActivity)) return;
+        if (walletType !== WalletType.BIP39) return;
+        this._initializeKeyguardClient();
```

Comment:
I would invert this condition like this. A positive condition is somehow easier to read than a negative one.
```suggestion
        if (walletType === WalletType.BIP39) this._initializeKeyguardClient();
```